pkg/yurthub/multiplexer/manager.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package multiplexer
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/generic/registry"
+	kstorage "k8s.io/apiserver/pkg/storage"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	ystorage "github.com/openyurtio/openyurt/pkg/yurthub/multiplexer/storage"
+)
+
+var NamespaceKeyFunc = func(obj runtime.Object) (string, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return "", err
+	}
+
+	return registry.NamespaceKeyFunc(request.WithNamespace(request.NewContext(), accessor.GetNamespace()), "", accessor.GetName())
+}
+
+var NoNamespaceKeyFunc = func(obj runtime.Object) (string, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return "", err
+	}
+
+	return registry.NoNamespaceKeyFunc(request.WithNamespace(request.NewContext(), accessor.GetNamespace()), "", accessor.GetName())
+}
+
+type MultiplexerManager interface {
+	ResourceCacheConfig(gvr *schema.GroupVersionResource) (*ResourceCacheConfig, error)
+	ResourceCache(gvr *schema.GroupVersionResource) (Interface, func(), error)
+}
+
+type multiplexerManager struct {
+	restStoreManager    ystorage.StorageManager
+	discoveryClient     discovery.DiscoveryInterface
+	cacheMap            map[string]Interface
+	cacheConfigMap      map[string]*ResourceCacheConfig
+	cacheDestroyFuncMap map[string]func()
+}
+
+func NewShareCacheManager(
```

Comment:
how about rename function name `NewShareCacheManager` to `NewRequestsMultiplexerManager`?

Reply:
done